### ISSUE TITLE
[Bug]: default sharing permission not working

### ISSUE THREAD

**Issue by keomabrun** [nextcloud/server#38149]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

The "Default share permissions" does not seem to be working.
Maybe I just don't understand what it is supposed to do.

I am not sure I understand the following line from [the documentation](https://docs.nextcloud.com/server/latest/admin_manual/configuration_files/file_sharing_configuration.html)

> With Default share permissions you are able to set the default permissions for user-shares (Create, Change, Delete and Reshare) without forcing them.

Does it mean that existing link permissions won't change, but new link will have the default permission set ? If yes, this is what I expect.

### Steps to reproduce

1. Go to Settings -> Administration Settings -> Sharing
2. Check all the check-boxes under "Default share permissions"
![image](https://github.com/nextcloud/server/assets/11633612/b9726bd7-5868-4410-bc5d-d0b16af3c0d6)
3. Go to the file explorer
4. Create a new link, and then open the link settings
![image](https://github.com/nextcloud/server/assets/11633612/7221b7ec-f229-4d4c-8e8a-fd14013039ad)

### Expected behavior

I would expect that the new link is created with the default permission that are set in the Administration Settings

### Installation method

Community Manual installation with Archive

### Nextcloud Server version

25

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 7.4

### Web server

Nginx

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost",
            "test3.girofle.cloud"
        ],
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "25.0.2.3",
        "overwrite.cli.url": "https:\/\/test3.girofle.cloud",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "default_language": "fr",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "default_phone_region": "fr",
        "maintenance": false,
        "loglevel": 2,
        "upgrade.disable-web": "true"
    }
}
```

### List of activated Apps

```shell
Enabled:
  - activity: 2.17.0
  - calendar: 4.2.0
  - circles: 25.0.0
  - cloud_federation_api: 1.8.0
  - comments: 1.15.0
  - contactsinteraction: 1.6.0
  - dashboard: 7.5.0
  - dav: 1.24.0
  - federatedfilesharing: 1.15.0
  - federation: 1.15.0
  - files: 1.20.1
  - files_pdfviewer: 2.6.0
  - files_rightclick: 1.4.0
  - files_sharing: 1.17.0
  - files_trashbin: 1.15.0
  - files_versions: 1.18.0
  - logreader: 2.10.0
  - lookup_server_connector: 1.13.0
  - nextcloud_announcements: 1.14.0
  - notifications: 2.13.1
  - notify_push: 0.5.2
  - oauth2: 1.13.0
  - onlyoffice: 7.6.8
  - password_policy: 1.15.0
  - photos: 2.0.1
  - privacy: 1.9.0
  - provisioning_api: 1.15.0
  - recommendations: 1.4.0
  - related_resources: 1.0.3
  - serverinfo: 1.15.0
  - settings: 1.7.0
  - sharebymail: 1.15.0
  - spreed: 15.0.3
  - support: 1.8.0
  - survey_client: 1.13.0
  - systemtags: 1.15.0
  - text: 3.6.0
  - theming: 2.0.1
  - twofactor_backupcodes: 1.14.0
  - updatenotification: 1.15.0
  - user_status: 1.5.0
  - viewer: 1.9.0
  - weather_status: 1.5.0
  - workflowengine: 2.7.0
Disabled:
  - admin_audit
  - bruteforcesettings
  - encryption
  - files_external
  - suspicious_login
  - twofactor_totp
  - user_ldap
```

### Nextcloud Signing status

```shell
No errors have been found.
```

**Comment by JSoko** [comment]
Any news about this FR?

Would be great to be able to set the standard permissions for public shares...
At the moment, I have to set Reading and Upload manually on every new share...

**Comment by flemmingss** [comment]
I agree with @JSoko . This is something that we should be able to set. And If I am not remembering wrong, this function also applied to public shares in earlier versions of Nextcloud?
([my thread](https://help.nextcloud.com/t/how-do-i-set-default-sharing-permissions/179718/2))

**Comment by flemmingss** [comment]
By the way, after a little more testing, looks like these settings don't apply for sharing with users either. May be a bug then?

**Comment by solracsf** [comment]
Please do a screencast and share it. These permission are just 'defaults', they can be override by users;

**Comment by flemmingss** [comment]
> Please do a screencast and share it. These permission are just 'defaults', they can be override by users;

I know, see:
![image](https://github.com/nextcloud/server/assets/25674067/bb88884a-0e78-420a-b1b6-7f26bb4c1e2b)
